### PR TITLE
#57 Clean up

### PR DESCRIPTION
**Pull request by Konoplastiy**

### CODE REVIEW

File: src/main/java/com/gachigang/ontherun/persistence/entity/Product.java [hunk 1]
```diff
@@ -32,7 +30,7 @@ public class Product {
 
     private String name;
     private Double price;
-    private LocalDateTime cookedAt;
+    private Integer cookingTime;
```
Comment (AlexandrVikhristyk) [inline src/main/java/com/gachigang/ontherun/persistence/entity/Product.java:33]: Bad idea, we already have discussed why

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class RefreshTokenService {
+
+    private final JwtService jwtService;
+    private final UserRepository userRepository;
+    private final TokenRepository tokenRepository;
+
+    public void refreshToken(
```
Comment (AlexandrVikhristyk) [inline src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:28]: In this method u do a lot of changes in DB, I believe it should be wrapped up with Transaction

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                var authResponse = AuthenticationResponse.builder()
+                        .accessToken(accessToken)
+                        .refreshToken(refreshToken)
+                        .build();
+                new ObjectMapper().writeValue(response.getOutputStream(), authResponse);
+            }
+        }
+    }
+
+    public void revokeAllUserTokens(User user) {
```
Comment (AlexandrVikhristyk) [inline src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:56]: Transactional

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        if (validUserTokens.isEmpty())
+            return;
+        validUserTokens.forEach(token -> {
+            token.setExpired(true);
+            token.setRevoked(true);
+        });
+        tokenRepository.saveAll(validUserTokens);
+    }
+
+    public void saveUserToken(User user, String jwtToken) {
```
Comment (AlexandrVikhristyk) [inline src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:67]: Transactional

**Review by AlexandrVikhristyk — COMMENTED** [review]
A couple of comments

File: src/main/java/com/gachigang/ontherun/persistence/entity/Product.java (hunk 1 above)
Comment (Konoplastiy) [reply src/main/java/com/gachigang/ontherun/persistence/entity/Product.java:33]: Ops, I will change the type from Integer to Period.

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java (hunk 2 above)
Comment (Konoplastiy) [reply src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:28]: I will add the Transaction annotation above this method and others in this service.

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java (hunk 3 above)
Comment (Konoplastiy) [reply src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:56]: Done

File: src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java (hunk 4 above)
Comment (Konoplastiy) [reply src/main/java/com/gachigang/ontherun/service/RefreshTokenService.java:67]: Done

**Review by ivanpoltavskiy — APPROVED** [review]

**Pull request merged**